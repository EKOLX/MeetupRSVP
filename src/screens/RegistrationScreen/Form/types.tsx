import { InputsModel } from "../types";

export interface Props {
  values: InputsModel;
  errors: Partial<InputsModel>;
  onInputChange: (text: string, input: keyof InputsModel) => void;
  onError: (errorMessage: string | null, input: keyof InputsModel) => void;
  onSubmit: () => void;
}
