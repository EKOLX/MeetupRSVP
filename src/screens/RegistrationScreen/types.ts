export interface InputsModel {
  name: string;
  age: string;
  birthdate: string;
  profession: string;
  locality: string;
  guestsCount: string;
  address: string;
}

export interface Validation {
  valid: boolean;
  inputs: InputsModel;
}
