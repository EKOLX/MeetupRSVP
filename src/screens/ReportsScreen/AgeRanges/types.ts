import { AgeRange } from "../types";

export interface Props {
  ageRanges: Array<AgeRange>;
}
