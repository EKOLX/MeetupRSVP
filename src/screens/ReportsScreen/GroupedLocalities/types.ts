import { Locality } from "../types";

export interface Props {
  localities: Array<Locality>;
}
