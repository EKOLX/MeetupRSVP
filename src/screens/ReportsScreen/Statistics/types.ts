export interface Props {
  averageGroupSize: number;
  employedCount: number;
  studentCount: number;
}
