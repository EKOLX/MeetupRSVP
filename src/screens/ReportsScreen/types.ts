export interface AgeRange {
  ageRange: string;
  count: number;
}

export interface Locality {
  locality: string;
  count: number;
}
