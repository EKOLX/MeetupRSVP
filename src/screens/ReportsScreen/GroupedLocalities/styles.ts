import { StyleSheet } from "react-native";

import Colors from "../../../constants/Colors";

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    marginTop: 22,
  },
  item: {
    flexDirection: "row",
    justifyContent: "space-between",
    borderBottomWidth: 1,
    borderBottomColor: Colors.gray,
    padding: 4,
  },
  header: {
    fontSize: 16,
  },
});
