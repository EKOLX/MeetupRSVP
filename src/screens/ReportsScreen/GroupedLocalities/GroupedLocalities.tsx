import { FC } from "react";
import { FlatList, ListRenderItem, Text, View } from "react-native";

import { Locality } from "../types";
import { Props } from "./types";
import { styles } from "./styles";

const GroupedLocalities: FC<Props> = ({ localities }) => {
  const renderLocality: ListRenderItem<Locality> = ({ item }) => (
    <View style={styles.item}>
      <Text>{item.locality}</Text>
      <Text>{item.count}</Text>
    </View>
  );

  return (
    <View style={styles.container}>
      <Text style={styles.header}>Grouped by Localities</Text>
      <FlatList
        showsVerticalScrollIndicator={false}
        data={localities}
        renderItem={renderLocality}
      />
    </View>
  );
};

export default GroupedLocalities;
