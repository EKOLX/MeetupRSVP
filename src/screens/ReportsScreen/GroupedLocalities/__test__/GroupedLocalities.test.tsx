import { create } from "react-test-renderer";

import GroupedLocalities from "../GroupedLocalities";
import { Locality } from "../../types";

describe("GroupedLocalities component", () => {
  const localities: Array<Locality> = [
    {
      locality: "Prague",
      count: 3,
    },
    {
      locality: "Russia",
      count: 5,
    },
  ];

  const groupedLocalities = create(
    <GroupedLocalities localities={localities} />
  );

  it("should match snapshot", () => {
    const tree = groupedLocalities.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
