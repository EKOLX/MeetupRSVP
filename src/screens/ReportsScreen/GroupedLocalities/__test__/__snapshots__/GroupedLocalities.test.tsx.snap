// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupedLocalities component should match snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "marginTop": 22,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
      }
    }
  >
    Grouped by Localities
  </Text>
  <RCTScrollView
    data={
      [
        {
          "count": 3,
          "locality": "Prague",
        },
        {
          "count": 5,
          "locality": "Russia",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#BACDDB",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 4,
            }
          }
        >
          <Text>
            Prague
          </Text>
          <Text>
            3
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#BACDDB",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 4,
            }
          }
        >
          <Text>
            Russia
          </Text>
          <Text>
            5
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
